assumes/expression.go

```diff
@@ -0,0 +1,61 @@
+// Copyright 2021 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package assumes
+
+import "github.com/juju/version/v2"
+
+var (
+	_ Expression = (*FeatureExpression)(nil)
+	_ Expression = (*CompositeExpression)(nil)
+)
+
+// ExpressionType represents the type of an assumes expression.
+type ExpressionType string
+
+const (
+	AnyOfExpression ExpressionType = "any-of"
+	AllOfExpression ExpressionType = "all-of"
+)
+
+// Expression is an interface implemented by all expression types in this package.
+type Expression interface {
+	Type() ExpressionType
+}
+
+// VersionConstraint describes a constraint for required feature versions.
+type VersionConstraint string
+
+const (
+	VersionGTE VersionConstraint = ">="
+	VersionLT  VersionConstraint = "<"
```

Comment:
I understand why you don't want `=`, as that exposes locked files, but why no LTE, that would seem useful?

Reply:
This was debated in the spec review session. Typically, you know the bare minimum version (hence the `>=`) and you now the version where it _stops_ working (hence the `<`) here.